tests/test_singularity_fixes.py

```diff
@@ -0,0 +1,175 @@
+import pytest
+from sympy import (
+    Function,
+    Piecewise,
+    exp,
+    log,
+)
+
+from cellmlmanip.model import Quantity, Variable
+from cellmlmanip.parser import XmlNs
+from cellmlmanip.singularity_fixes import fix_singularity_equations, new_expr
+
+from . import shared
+
+
+def to_quant(val):
+    return Quantity(val, 'dimensionless')
+
+
+V = Variable(name='V', units='millivolt')
+Z = to_quant(6.2)
+U = to_quant(2) * V + to_quant(5.0)
+SP = (-(log(Z) + to_quant(5.0)) / to_quant(2))
+
+
+@pytest.fixture(scope='session')
+def expr1():
+    return (Z * V - Z * SP) / (exp(U) * -Z + to_quant(1.0))
+
+
+@pytest.fixture(scope='session')
+def expr2():
+    return (Z * V - Z * SP) / (exp(U) * Z - to_quant(1.0))
+
+
+class exp_(Function):
+
+    def _eval_is_real(self):
+        return self.args[0].is_real
+
+    def fdiff(self, argindex=1):
+        """
+        Returns the first derivative of this function.
+        """
+        assert argindex == 1
+        return self
+
+
+def test_no_singularity():
+    expr = (exp_(U) * -Z + to_quant(1.0)) / to_quant(25.0)
+    changed, new_ex = new_expr(expr, V)
+    assert str(new_expr(expr, V)) == str((False, expr))
+
+
+def test_no_singularity2():
+    expr = to_quant(1) / (to_quant(1) + exp_(to_quant(1.5669291338582676) - to_quant(0.07874015748031496) * V))
+    assert str(new_expr(expr, V)) == str((False, expr))
+
+
+def test_new_expr1(expr1):
+    print(str(new_expr(expr1, V)))
+    assert str(new_expr(expr1, V)) == ('(True, Piecewise(((-_-3.4122746960255230 + _V)*(-(_-3.4122746960255230*_6.2 - '
```

Comment:
Can you add a comment above each of these showing what the input and output equations are in a readable form?